fdb-relational-core/src/main/java/com/apple/foundationdb/relational/recordlayer/query/LogicalOperator.java

```diff
@@ -184,7 +185,7 @@ public static LogicalOperator generateAccess(@Nonnull Identifier identifier,
             return logicalOperatorCatalog.lookupTableAccess(identifier, alias, requestedIndexes, semanticAnalyzer);
         } else {
             final var correlatedField = semanticAnalyzer.resolveCorrelatedIdentifier(identifier, currentPlanFragment.getLogicalOperatorsIncludingOuter());
-            Assert.thatUnchecked(requestedIndexes.isEmpty(), ErrorCode.UNSUPPORTED_QUERY, () -> String.format("Can not hint indexes with correlated field access %s", identifier));
+            Assert.thatUnchecked(requestedIndexes.isEmpty(), ErrorCode.UNSUPPORTED_QUERY, () -> String.format(Locale.ROOT, "Can not hint indexes with correlated field access %s", identifier));
```

Comment:
Probably quite a few of these would be better off using string concatenation, but using the simple mechanical change of adding `Locale.ROOT` seems good for this broad change.

Reply:
Yes, I will propose a different PR to remove usage of `String.format` where applicable.

Admittedly, most (if not all) of the current usages of `String.format` in `relational` are in an exceptional code path; either in an assertion failure, or to formulate an exception message. I do not think optimizing exceptional code paths for performance is crucial (they should not be in a tight loop, stack unrolling is already _very_ expensive, ... etc).

Nevertheless, it is probably a good practice to avoid using `String.format` all together, and have a code analysis check that errors out when one uses it by mistake, considering how terrible its performance is.